doc/designs/ML-KEM.md

```diff
@@ -25,13 +25,13 @@ of macros to minimize source code duplication is used. Note that C++ templates a
 
 ML-KEM makes extensive use of SHA3 primitives, SHA3-256, SHA3-512, SHAKE256 and SHAKE128.
 To improve ML-KEM execution performance the EVP handles for these are pre-fetched during ML-KEM
-key initialisation and stored in an MLKEM_CTX object.
+key initialisation and stored in an ossl_ml_kem_ctx object.
```

Comment:
Type names are supposed to be uppercase.  I know this represents the current code but that code needs to change IMO.

Reply:
I also know that :-/

My intention (I thought in meetings with @vavroch2010 and @mattcaswell agreed-upon to close out a sprint) was to finish off/clean up this documentation task with a separate PR _after_ the code is finalized (https://github.com/openssl/openssl/pull/26054 has landed) and all "naming issues" have been resolved in the code. With his latest feedback @t-j-h however demanded this PR to be in line with the WIP which I also tried to satisfy now (invalidating the reviews and changing casing once again, I know). Trying to satisfy everyone is a bit challenging when reviews straddle (way) more than 1 sprint.

Reply:
Type names used in *PUBLIC* interfaces, are supposed to be in upper case. This is NOT a public interface. There are a few other internal type names that are not upper case. Admittedly, rare beasts:
```
include/crypto/ecx.h-} ecx_key_op_t;
include/crypto/evp.h-} evp_cipher_aead_asn1_params;
include/crypto/ml_kem.h-} ossl_ml_kem_vinfo;
include/crypto/ml_kem.h-} ossl_ml_kem_scalar;
include/internal/dane.h-} danetls_record;
include/internal/hashtable.h-} keyname;
include/internal/ktls.h:typedef struct tls_enable ktls_crypto_info_t;
include/internal/ktls.h:typedef struct tls_crypto_info_all ktls_crypto_info_t;
```
and some public ones:
```
include/openssl/ct.h.in-} ct_log_entry_type_t;
include/openssl/ct.h.in-} sct_version_t;
include/openssl/ct.h.in-} sct_source_t;
include/openssl/ct.h.in-} sct_validation_status_t;
include/openssl/ec.h-} point_conversion_form_t;
include/openssl/engine.h-} dynamic_MEM_fns;
include/openssl/engine.h-} dynamic_fns;
```

Anyway the `ctx` in question is about to be consolidated into the key, which is being renamed `ML_KEM_KEY`. The only question is whether the current state of the PR is sufficient as a phase-I working draft, to make it possible to start building hybrids, ..., that just need EVP support, and don't care about the internals.

Or, whether I should just push the cleaned up phase-II code for review.